src/main/java/com/faforever/moderatorclient/ui/main_window/LadderMapPoolController.java

```diff
@@ -329,4 +412,82 @@ public void onGeneratedMapButton() {
         selectedMap.setValue(mapPoolAssignmentMapper.map(mapPoolAssignment));
     }
 
+    public List<Integer> getFavorites() throws IOException {
+        ArrayList<Integer> list = Lists.newArrayList();
+
+        File pwd = new File("").getAbsoluteFile();
+        File file = new File(pwd.getAbsoluteFile() + File.separator + "favoriteMaps.txt");
+
+        if (!file.exists()) {
+            file.createNewFile();
+        }
+
+        Scanner reader = new Scanner(file);
+        while (reader.hasNextLine()) {
+            String line = reader.nextLine();
+            if (line != null && line.length() > 0) {
+                list.add(Integer.parseInt(line));
+            }
+        }
+        reader.close();
+        return list;
+    }
+
+    public void addFavorite(MapTableItemAdapter mapTableItemAdapter) {
+        System.out.println(mapTableItemAdapter.getId());
+        System.out.println(mapTableItemAdapter.getMap().getId());
+        if (favoritesCache.contains(Integer.parseInt(mapTableItemAdapter.getId()))) return;
```

Comment:
This whole integer parsing is EVERYWHERE. It seems much simpler to just use string everywhere.